cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DocumentStore.java

```diff
@@ -170,9 +167,15 @@ public void delete() throws IOException {
         if (!location.exists()) {
             String msg = String.format("Datastore %s doesn't exist on disk", location);
             logger.warning(msg);
-            throw new IOException(msg);
+            throw new DatastoreNotDeletedException(msg);
```

Comment:
I feel like a `FileNotFound` would be a better exception, although there is a part of me that thinks we should crash because the state isn't right. Calling `delete` twice seems like a programmer error than something that we should be recovering from. Thoughts?